app/activity_notifications/chat_message_sent.rb

```diff
@@ -3,15 +3,16 @@
 class ChatMessageSent < Noticed::Base
   deliver_by :database, format: :to_database, association: :notifications_as_recipient
 
-  param :contributor_id, :request_id, :user_id, :message_id
+  param :contributor_id, :request_id, :user_id, :message_id, :organization_id
```

Comment:
From a db normalization perspective we are denormalizing  the organization_id, if a request always belongs to one organization. This might already be the same for user_id and contributor_id as they might be deducted from the message itself.

Reply:
So, we discussed this by phone, but I'll just write here for our future selves. In the case of the notifications, we added the organization_id as a param and save it to a activity_notification record in the database because we have admin who do not belong to any organization and we have potential clients that belong to multiple organizations so we use the organization id to filter the notifications to the organization that is set by the route.

Since we don't actually load the request itself, but simply pass in the request_id, it would be more complicated to derive the contributor, message, user, and organization from the activity_notification record without passing in these ids. 

What we do do, is that when we create the activity_notification record, we do use the request, sender, organization, etc. from the message record, which I guess addresses your concern that they somehow don't match. See :point_down: 

```rb
MessageReceived.with(
  contributor_id: sender_id,
  request_id: request.id,
  message_id: id,
  organization_id: organization.id
).deliver_later(organization.users + User.admin.all)
```